### ISSUE TITLE
Lack of accessibility

### ISSUE THREAD

**Issue by chkpnt** [bonprix/vaadin-combobox-multiselect#14]
The standard `ComboBox` from Vaadin works as follows:

1. When the textbox is focused, it's content is selected. A screenreader reads the selected text
2. When an element from the suggestion popup is selected (highlighted), the textbox's content is immediately updated with the selected element. A screenreader observes the change and reads it.

As the `ComboBoxMulitselect` works different by design, a screenreader is stranded. I find the following behaviour useful:

1. When the user navigates using the arrow keys, the screenreader should read the element form the suggetion popup and it's activation state
2. The screenreader reads the element's position in the suggention popup ("entry 4 of 20").

Unfortunately, Jaws reads `aria-setsize` and `aria-posinset` only if they are set in an element with `role="listitem"`, whereas the [aria specs](https://www.w3.org/WAI/PF/aria/states_and_properties#aria-setsize) allows `role="option"`, too. And I have to use the latter role in order to use `aria-checked` on the `td` element. Setting `aria-activedescendant` to the checkbox instead of the `td` element is not an option, as this leads to focused checkboxes and a broken styling.

I've done some first work in my branch [Barrierefreiheit](https://github.com/bonprix/vaadin-combobox-multiselect/compare/master...chkpnt:Barrierefreiheit). What do you think?

**Comment by chkpnt** [comment]
In case of a discussion: I'm afk 'till next week, so please bear with me.

**Comment by thorbenvh8** [comment]
I will look into this next week once youre so we can talk about this. Sounds very interesting. Is your pull request completly implementing the feature?

**Comment by chkpnt** [comment]
Yes, from my side, this feature is implemented. The only reason I haven't formulated this as a PR is to be able to do a rebase / squash on my branch in case further changes are arising from this thread.

To get a better impression, I'm attaching two screencasts, in which you see and hear (in German) the behaviour of NVDA (wich is reading the entry's position in contrast to Jaws) before and after the proposed changes: [current state.zip](https://github.com/bonprix/vaadin-combobox-multiselect/files/407056/current.state.zip), [with enhanced acccessibility support.zip](https://github.com/bonprix/vaadin-combobox-multiselect/files/407058/with.enhanced.acccessibility.support.zip)

**Comment by thorbenvh8** [comment]
looks good to me :) very interesting.
Seemed kind of easy to implement, what is the difference to the libraries from gwt.aria?
If you want you can make this a pull request :)

**Comment by chkpnt** [comment]
I don't know the gwt-aria-classes in detail, I'm using them only to prevent String-based calls like `getElement().setPropertyString("aria-label", ...)` or getElement().setAttribute("aria-checked", "true"). As far as I saw in other people's code, statements like `State.CHECKED.set(menuItem.getElement(), CheckedValue.TRUE)` seem to be the gwt-way to do aria-stuff.

The pull request is on its way :-)

**Comment by thorbenvh8** [comment]
I will build this into a version for vaadin repository in the next days.

Thanks :)
